.github/workflows/release.yml

```diff
@@ -1,45 +1,41 @@
-name: 'Release Charts'
+name: Release
 
 on:
   push:
     branches:
-      - 'main'
+      - main
 
-jobs:
-  release:
+permissions:
+  contents: write
+  packages: write
+  id-token: write
+  checks: read
 
-    runs-on: 'ubuntu-latest'
+defaults:
+  run:
+    working-directory: ./
 
-    permissions:
-      contents: 'write'
-      packages: 'write'
-      id-token: 'write'
+jobs:
+  publish-package:
+    runs-on: ubuntu-latest
+    name: To Artifactory
 
     steps:
-      - id: 'checkout'
-        uses: 'actions/checkout@v5'
+      - name: Harden runner
+        uses: step-security/harden-runner@0634a2670c59f64b4a01f0f96f84700a4088b9f0 # v2.12.0
         with:
-          fetch-depth: "0"
+          egress-policy: audit
 
-      - id: 'configure-git'
-        run: |
-          git config user.name "$GITHUB_ACTOR"
-          git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
+      - name: Checkout code
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
```

Comment:
Why a sha version?

Reply:
For provenance. This is just to ensure that if someone changes the tag upstream, our CI doesn't use that version. This allows us to pin. Normally, these are all automated. In most repos, we are using step-security to ensure that the CI pipeline is secure. https://github.com/zitadel/client-java/blob/beta/.github/workflows/scorecard.yml

This runs periodically and autoamtically checks the pipeline. They have an app, that auto creates PRs for the corresponding version and SHA to ensure that the pipelines are robust.